packages/asset-swapper/src/utils/swap_quote_calculator.ts

```diff
@@ -208,199 +130,159 @@ function calculateSwapQuote(
             return {
-                totalTakerTokenAmount: totalTakerTokenAmount.plus(takerFillAmount),
-                totalZrxAmount: totalZrxAmount.plus(takerFeeAmount),
-                remainingmakerAssetFillAmount: BigNumber.max(
+                totalMakerAssetAmount: totalMakerAssetAmount.plus(makerFillAmount),
+                totalTakerAssetAmount: totalTakerAssetAmount.plus(takerAssetAmount),
+                totalFeeTakerAssetAmount: totalFeeTakerAssetAmount.plus(feeTakerAssetAmount),
+                remainingMakerAssetFillAmount: BigNumber.max(
                     constants.ZERO_AMOUNT,
-                    remainingmakerAssetFillAmount.minus(makerFillAmount),
+                    remainingMakerAssetFillAmount.minus(makerFillAmount),
                 ),
             };
         },
         {
-            totalTakerTokenAmount: constants.ZERO_AMOUNT,
-            totalZrxAmount: constants.ZERO_AMOUNT,
-            remainingmakerAssetFillAmount: makerAssetBuyAmount,
+            totalMakerAssetAmount: constants.ZERO_AMOUNT,
+            totalTakerAssetAmount: constants.ZERO_AMOUNT,
+            totalFeeTakerAssetAmount: constants.ZERO_AMOUNT,
+            remainingMakerAssetFillAmount: makerAssetBuyAmount,
         },
     );
-    return [result.totalTakerTokenAmount, result.totalZrxAmount];
+    return {
+        feeTakerAssetAmount: result.totalFeeTakerAssetAmount,
+        takerAssetAmount: result.totalTakerAssetAmount,
+        totalTakerAssetAmount: result.totalFeeTakerAssetAmount.plus(result.totalTakerAssetAmount),
+        makerAssetAmount: result.totalMakerAssetAmount,
+        protocolFeeInEthAmount: utils.calculateWorstCaseProtocolFee(prunedOrders, gasPrice),
+    };
 }
 
-function findMakerTokenAmountReceivedAndZrxAmountNeededToSellAsset(
-    ordersAndFillableAmounts: OrdersAndFillableAmounts,
-    takerAssetSellAmount: BigNumber,
-): [BigNumber, BigNumber] {
-    const { orders, remainingFillableMakerAssetAmounts } = ordersAndFillableAmounts;
-    const result = _.reduce(
-        orders,
-        (acc, order, index) => {
-            const { totalMakerTokenAmount, totalZrxAmount, remainingTakerAssetFillAmount } = acc;
-            const remainingFillableMakerAssetAmount = remainingFillableMakerAssetAmounts[index];
-            const remainingFillableTakerAssetAmount = orderCalculationUtils.getTakerFillAmount(
-                order,
-                remainingFillableMakerAssetAmount,
-            );
-            const takerFillAmount = BigNumber.min(acc.remainingTakerAssetFillAmount, remainingFillableTakerAssetAmount);
-            const makerFillAmount = orderCalculationUtils.getMakerFillAmount(order, takerFillAmount);
-            const takerFeeAmount = orderCalculationUtils.getTakerFeeAmount(order, takerFillAmount);
+function getTakerAssetAmountBreakDown(
+    order: PrunedSignedOrder,
+    takerAssetAmountWithFees: BigNumber,
+): { feeTakerAssetAmount: BigNumber; takerAssetAmount: BigNumber } {
```

Comment:
The main reasoning behind the choice of referring to this amount as `feeTakerassetAmount` is because it reflects the amount of `takerAssetAmount` that will be allocated to pay off the `takerFee`; since the `takerFee` can be denominated in maker or taker asset, `takerFeeAssetAmount` is not equatable to `feeTakerAssetAmount`